.github/workflows/papi_framework.yml

```diff
@@ -0,0 +1,78 @@
+name: papi framework
+
+on:
+  pull_request:
+    # run CI if framework receives an update excluding individual 
+    # components and counter analysis toolkit
+    paths:
+      - 'src/**'
+      - 'src/components/*'
+      - '!src/components/*/**'
+      - '!src/counter_analysis_toolkit/**'
+  # allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  # build PAPI with multiple components to simulate a users' workflow
+  # rocm, rocm_smi, powercap_ppc, rapl, sensors_ppc, infiniband, lustre, and mx should all
+  # be disabled upon the build finishing
+  papi_components_comprehensive:
+    strategy:
+      matrix:
+        components: [cuda nvml rocm rocm_smi powercap powercap_ppc rapl sensors_ppc infiniband net appio io lustre stealtime coretemp lmsensors mx sde]
+        debug: [yes, no]
+        shlib: [with, without]
+      fail-fast: false
+    runs-on: [self-hosted, cpu_intel, gpu_nvidia]
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test
+        run: .github/workflows/ci_papi_framework.sh "${{matrix.components}}" ${{matrix.debug}} ${{matrix.shlib}}
+  # build PAPI only with the amd components, as they will not be active in the above comprehensive job
+  papi_components_amd:
+    strategy:
+      matrix:
+        components: [rocm rocm_smi]
+        debug: [yes, no] 
+        shlib: [with, without]
+      fail-fast: false
+    runs-on: [self-hosted, gpu_amd]
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v4
+      - name: Test
+        run: .github/workflows/ci_papi_framework.sh "${{matrix.components}}" ${{matrix.debug}} ${{matrix.shlib}}
+  # build PAPI only with the infiniband component, as it will not be active in the above comprehensive job
+  papi_component_infiniband:
+    strategy:
+      matrix:
+        components: [infiniband]
+        debug: [yes, no] 
+        shlib: [with, without]
+      fail-fast: false
+    runs-on: [self-hosted, guyot]
```

Comment:
You can change "guyot" to "infiniband" here.

Reply:
I have made that update in the latest push.